src/msg/async/ProtocolV2.cc

```diff
@@ -2893,5 +2962,47 @@ CtPtr ProtocolV2::send_reconnect_ok() {
 
   INTERCEPT(14);
 
+  if (HAVE_MSGR2_FEATURE(peer_supported_features, COMPRESSION)) {
+    state = COMPRESSION_ACCEPTING;
+    return WRITE(reconnect_ok, "reconnect ok", read_frame);  
+  }
+
   return WRITE(reconnect_ok, "reconnect ok", server_ready);
 }
+
+
+CtPtr ProtocolV2::handle_compression_request(ceph::bufferlist &payload) {
+  if (state != COMPRESSION_ACCEPTING) {
+    lderr(cct) << __func__ << " state changed!" << dendl;
```

Comment:
Currently - should never happen. If it does, it means there's a bug in handshake between client and server. 

This method can only be reached if both client and server have been upgraded and are aware of the compression feature (otherwise, the Tag wasn't recognized and would have resulted in server fault).  ​
That being said, we don't to accidentaly reach this method randomly. For example, if we reach this method when server is already sending messages, we might send a compressed message to the client before it had a chance to setup its' compression handlers.